starport-blog/01-index.md

```diff
@@ -0,0 +1,350 @@
+func (msg MsgCreatePost) ValidateBasic() error {
+  if msg.Creator.Empty() {
+    return sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, "creator can't be empty")
+  }
+  return nil
+}
+```
+
+Going back to `GetCmdCreatePost`, you created a new `MsgCreatePost` message and broadcasted it with `GenerateOrBroadcastMsgs`.
+
+After being broadcasted, the messages are processed by handlers.
+
+## x/blog/handler.go
+
+```go
+import (
+  // Existing imports...
+  "github.com/example/blog/x/blog/types"
+)
+```
+
+You should already have `func NewHandler` defined which lists all available handlers. Modify it to include a new function called `handleMsgCreatePost`.
+
+```go
+    switch msg := msg.(type) {
+    case MsgCreatePost:
+      return handleMsgCreatePost(ctx, k, msg)
+    default:
+```
+
+Now let’s define `handleMsgCreatePost`:
+
+```go
+func handleMsgCreatePost(ctx sdk.Context, k Keeper, msg MsgCreatePost) (*sdk.Result, error) {
+  var post = types.Post{
+    Creator: msg.Creator,
+    ID:      msg.ID,
+    Title:   msg.Title,
+  }
+  k.CreatePost(ctx, post)
+  return &sdk.Result{Events: ctx.EventManager().Events()}, nil
+}
+```
+
+In this handler you create a `Post` object (post type was defined in the very first step). You populate the post object with creator and title from the message (`msg.Creator` and `msg.Title`) and generate a unique ID with `uuid.New().String()`. Make sure to add `"github.com/google/uuid"` to the import at the top of this file.
+
+After creating a post object with creator, ID and title, message handler calls `k.CreatePost(ctx, post)`. “k” stands for Keeper, an abstraction used by the SDK that writes data to the store. Let’s define `CreatePost` keeper function.
+
+## x/blog/keeper/keeper.go
+
+```go
+func (k Keeper) CreatePost(ctx sdk.Context, post types.Post) {
+  store := ctx.KVStore(k.storeKey)
+  key := []byte(types.PostPrefix + post.ID)
+  value := k.cdc.MustMarshalBinaryLengthPrefixed(post)
+  store.Set(key, value)
+}
+```
+
+`CreatePost` creates a key by concatenating a post prefix with an ID. If you look back at how our store looks, you’ll notice keys have prefixes `post-0bae9f7d-20f8-4b51-9d5c-af9103177d66` in this case `post-` . The reason for this is you have one store, but you might want to keep different types of objects in it, like posts and users. Prefixing keys with `post-` and `user-` allows you to share one namespace between different types of objects.
+
+## x/blog/types/key.go
+
+To define the post prefix add the following code:
+
+```go
+package types
+
+const (
+  // Other constants...
+  // PostPrefix is used for keys in the KV store
+  PostPrefix = "post-"
+)
+```
+
+## x/blog/types/codec.go
+
+Finally, `store.Set(key, value)` writes our post to the store.
+Two last things to do is tell our encoder how our `MsgCreatePost` is converted to bytes.
+
+```go
+func RegisterCodec(cdc *codec.Codec) {
+  cdc.RegisterConcrete(MsgCreatePost{}, "blog/CreatePost", nil)
+}
+```
+
+## x/blog/alias.go
```

Comment:
alias.go is no longer needed and not in the module spec. we can remove it from starport?

Reply:
Is there a doc I can read about this? Because alias.go is still in the Nameservice, so I figured it's still the best practice.